### PR TITLE
pr(archai): Adds support for Huggingface's Transformer-XL and configurable ONNX exports.

### PR DESCRIPTION
**Pull request by gugarosa**
This PR introduces the following changes:

- Fixes a problem where integer-based arguments were not being properly loaded in `HfGPT2`;
- Adds support for `HfTransfoXL`, which is Huggingface's implementation of the Transformer-XL;
- Cleans unnecessary imports;
- Adds license headers to new files;
- Fixes a bug when loading an `ArchaiModel` and not having PyTorch compiled with CUDA;
- Changes `models/model_list.py` to `models/available_models.py`;
- Allows for configurable ONNX exports;
- Allows for configurable ONNX optimizations.

### CODE REVIEW

**Comment by gugarosa** [comment]
Tests are being run to check for possible breaks or bugs.

File: archai/nlp/nvidia_transformer_xl/models/archai_model.py [hunk 1]
```diff
@@ -23,27 +25,25 @@ def get_n_params(self):
         return sum([p.nelement() for p in self.parameters()])
 
     @staticmethod
-    def load_model(model_cls: Type['ArchaiModel'], path:str, model:Optional['ArchaiModel']=None, on_cpu:Optional[bool]=False) -> Tuple['ArchaiModel', dict, dict]:
+    def load_model(model_cls: Type['ArchaiModel'], path:str, model:Optional['ArchaiModel']=None, on_cpu:Optional[bool]=False, on_export:Optional[bool]=False) -> Tuple['ArchaiModel', dict, dict]:
```
Comment (sytelus) [inline archai/nlp/nvidia_transformer_xl/models/archai_model.py:28]: Rename to for_export

File: archai/nlp/nvidia_transformer_xl/onnx/export_torch_to_onnx.py [hunk 2]
```diff
@@ -20,6 +20,12 @@ def parse_args():
                         type=str,
                         help='Path to the output ONNX model file.')
 
+    parser.add_argument('-model_type',
```
Comment (sytelus) [inline archai/nlp/nvidia_transformer_xl/onnx/export_torch_to_onnx.py:23]: We probably don't need "-" (use "--")

File: archai/nlp/nvidia_transformer_xl/onnx/onnx_utils/configs.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        """Initializes the configuration.
+
+        Args:
+            model_config: Model configuration.
+
+        """
+
+        super().__init__(model_config)
+
+        if self.config['attn_type'] == 0:
```
Comment (sytelus) [inline archai/nlp/nvidia_transformer_xl/onnx/onnx_utils/configs.py:67]: comment for magic values

File: archai/nlp/nvidia_transformer_xl/onnx/onnx_utils/configs.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        self.config['model_type'] = 'transfo-xl'
+
+    @property
+    def mockups(self) -> Dict[str, Any]:
+        """Defines the mockups (inputs) to be used when exporting to ONNX.
+        
+        """
+
+        return {
+            'input_ids': torch.randint(0, self.config['n_token'], (1, 32)),
```
Comment (sytelus) [inline archai/nlp/nvidia_transformer_xl/onnx/onnx_utils/configs.py:81]: why 32

File: archai/nlp/nvidia_transformer_xl/onnx/onnx_utils/configs.py [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+            'past_key_values': tuple([torch.zeros(self.config['past_key_values'], 1, self.config['n_head'], 32, self.config['d_head']) for _ in range(self.config['n_layer'])])
+        }
+
+    @property
+    def inputs(self) -> OrderedDict:
+        """Defines the inputs and their shapes to be used when exporting to ONNX.
+        
+        """
+
+        pasts = [(f'past_{i}', {0: str(self.config['past_key_values']), 1: 'batch_size', 2: str(self.config['n_head']), 3: 'past_seq_len', 4: str(self.config['d_head'])}) for i in range(self.config['n_layer'])]
```
Comment (sytelus) [inline archai/nlp/nvidia_transformer_xl/onnx/onnx_utils/configs.py:91]: comment: where did this came from

File: archai/nlp/nvidia_transformer_xl/onnx/onnx_utils/forward.py [hunk 6]
```diff
@@ -40,6 +40,7 @@ def forward_with_probs(self,
 
     # Calculates the output predictions/probabilities
     preds = self.crit(hidden_preds)
+    preds = preds.view(input_ids.size(1), -1) if preds is not None else None
```
Comment (sytelus) [inline archai/nlp/nvidia_transformer_xl/onnx/onnx_utils/forward.py:43]: Comment on what this line does

**Review by sytelus — APPROVED** [review]

**Pull request merged**